packages/core/src/sharedWishlists/redux/middlewares/updateSharedWishlistUponItemsChanges.js

```diff
@@ -0,0 +1,41 @@
+import {
+  ADD_ITEM_TO_WISHLIST_SUCCESS,
+  DELETE_WISHLIST_ITEM_SUCCESS,
+  UPDATE_WISHLIST_ITEM_SUCCESS,
+} from '../../../wishlists/redux/actionTypes';
+import { doUpdateSharedWishlist } from '../actions';
+import { getSharedWishlistId } from '../selectors';
+import { getUser, getUserIsGuest } from '../../../entities/redux/selectors';
+import { putSharedWishlist as putSharedWishlistClient } from '../../client';
+/**
+ * Middleware to update the shared wishlist if a wishlist item is added, edited or deleted.
+ *
+ * @function updateSharedWishlistUponItemsChanges
+ * @memberof module:sharedWishlists/middlewares
+ *
+ * @param {object} store - Redux store at the moment.
+ *
+ * @returns {Function} Redux middleware.
+ */
+export default store => next => action => {
+  if (
+    action.type ===
+    (ADD_ITEM_TO_WISHLIST_SUCCESS ||
+      UPDATE_WISHLIST_ITEM_SUCCESS ||
+      DELETE_WISHLIST_ITEM_SUCCESS)
+  ) {
+    const user = getUser(store.getState());
+    const isGuestUser = getUserIsGuest(user);
+
+    if (!isGuestUser) {
+      const sharedWishlistId = getSharedWishlistId(store.getState());
+
+      const updateSharedWishlist = doUpdateSharedWishlist(
```

Comment:
This can be done outside the middleware as it will be always the same value.

Reply:
What I meant to be outside the middleware was the creation of the thunk as it does not depend on anything else, except the client:

```suggestion
      const updateSharedWishlist = doUpdateSharedWishlist(putSharedWishlistClient); <--- This can be outside the middleware function as it does not depend on anything inside the function and is static.
```